### PR TITLE
fix: remove sarif property bag

### PR DESCRIPTION
**Pull request by brendaHuang**
[#496](https://github.com/rehearsal-js/rehearsal-js/issues/496) 

Tried using the Fixes property of sarif, but Fixes won't show in sarif reader. So I am sticking to property bag, which is more straightforward. 

Please refer to the attachments for "after" and "before" screenshots.

### CODE REVIEW

**Comment by brendaHuang** [comment]
Screen Shot 2022-12-08 at 6 47 51 PM

**Comment by brendaHuang** [comment]
Screen Shot 2022-12-08 at 6 54 40 PM

File: packages/reporter/src/formatters/sarif-formatter.ts [hunk 1]
```diff
@@ -124,30 +124,27 @@ export class SarifFormatter {
...
+          codeFixAction: file.codeFixAction || undefined,
+        };
       }
     });
 
     return {
       locations,
       relatedLocations,
-      fixes,
+      codeFix,
```
Comment (lynchbomb) [inline packages/reporter/src/formatters/sarif-formatter.ts:147]: want to be sure we want to change the return type to codeFix object vs an array of fixes. do we not need all the fixes anymore in this sarif entry?

**Review by lynchbomb — CHANGES_REQUESTED** [review]

File: packages/reporter/src/formatters/sarif-formatter.ts (hunk 1 above)
Comment (brendaHuang) [reply packages/reporter/src/formatters/sarif-formatter.ts:147]: I was debating this as well. If we have an array of fixes, the sarif reader won't display the fix object in the array. So this fixes or codeFixes has to be an object that's one level deep. One workaround is to append the file name to each of the 'filename', 'newCode', 'oldCode', 'codeFixAction', for example, 'fileName_xx', 'newCode_xx', 'oldCode_xx', 'codeFixAction_xx'. This way, if we have multiple files affected, we can scale to express that, though the current fix in this PR works for our current use cases. If we have more than multiple codefixes within 1 file, we can do 'filename_xx_1', 'newCode_xx_1', 'oldCode_xx_1, 'codeFixAction_xx_1'. Please let me know if this works. @lynchbomb

Comment (lynchbomb) [reply packages/reporter/src/formatters/sarif-formatter.ts:147]: > though the current fix in this PR works for our current use cases
ok lets keep it as implemented in this PR and expand when needed.

**Review by lynchbomb — APPROVED** [review]

**Pull request merged**